src/state/dashboard/actions.js

```diff
@@ -0,0 +1,180 @@
+import moment from 'moment';
+
+function getTimePoints(timeRange = 'week', timeUnit = 'day') {
+  const gtDate = moment().startOf(timeRange),
+    timePoints = [];
+  let datePoint = gtDate.clone().startOf(timeUnit);
+
+  while (datePoint.isSameOrBefore(moment())) {
+    timePoints.push(datePoint.clone());
+    datePoint.add(1, timeUnit);
+  }
+
+  return timePoints;
+}
+
+function getTimeUnit(range) {
+  switch (range) {
+    case 'day': {
+      return 'hour';
+    }
+    case 'year': {
+      return 'month';
+    }
+    default: {
+      return 'day';
+    }
+  }
+}
+
+function createTimeQueries(
+  endPoint = '/jobs/survey/',
+  timePoints = [],
+  isCumulative = false,
+  limit = 1000000000
+) {
+  const promiseArray = timePoints.map(async (datePoint, i) => {
+    const gte = isCumulative
+      ? timePoints[0].utc().format()
+      : timePoints[i].utc().format();
+    const lte = timePoints[i + 1]
+      ? timePoints[i + 1].utc().format()
+      : moment()
+          .utc()
+          .format();
+
+    const response = await fetch(
+      `${endPoint}?created_at__gte=${gte}&created_at__lte=${lte}&limit=${limit}`
+    );
+
+    if (response.status === 200) {
+      return await response.json();
+    } else {
```

Comment:
I guess you can delete lines 52 and 58 to make the code a little more compact (up to you).